File: 56-merge-intervals/merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        if (intervals.size() == 1){
            return intervals;
        }
        vector<vector<int>> merged_arr;
        std::sort(intervals.begin(), intervals.end());
        for (const std::vector<int>& entry: intervals){
            if (merged_arr.empty()){
                merged_arr.push_back(entry);
            }
            else {
                vector<int>& last =  merged_arr.back();
                if (entry[0] > last[1]){
                    merged_arr.push_back(entry);
                } 
                else {
                    if (entry[1] > last[1]){
                        last[1] = entry[1];
                    }
                }
            }
        }
        return merged_arr;
    }
};

File: 56-merge-intervals/merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        if len(intervals) == 1:
            return intervals
        intervals.sort()
        sorted_int = []
        sorted_int.append(intervals[0])
        for j in range(1, len(intervals)):
            if sorted_int[-1][1] >= intervals[j][0]:
                sorted_int[-1][0] = min(sorted_int[-1][0], intervals[j][0])
                sorted_int[-1][1] = max(sorted_int[-1][1], intervals[j][1])
            else:
                sorted_int.append(intervals[j])
        return sorted_int
                